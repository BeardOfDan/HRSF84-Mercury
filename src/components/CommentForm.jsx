import React from 'react';

const CommentForm = ({ onSubmit, onChangeName, onChangeText, onChangeCategory, name, text, category = null }) => {
  const inputStyle = {
    'border': '1px solid black',
    'padding': '5px',
    'margin': '5px'
  };

  return (
    <div className='form-comment container'>
      <form onSubmit={onSubmit}>
        {(category !== null) ?
          (
            <select onChange={onChangeCategory}>
              <option value="Quote">Quote</option>
              <option value="To Everyone">To Everyone</option>
              <option value="To Staff">To Staff</option>
              <option value="Meme">Meme</option>
            </select>
          ) : ''
        }
        <input style={inputStyle} type="text" name='name' placeholder='name' onChange={onChangeName} value={name} />
        <input style={inputStyle} type="text" name='comment' placeholder='comment' onChange={onChangeText} value={text} />
        <input type='submit' value='Submit Comment' />
      </form>
    </div>
  )
};

export default CommentForm;

