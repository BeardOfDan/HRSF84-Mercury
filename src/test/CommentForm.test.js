import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import CommentForm from '../components/CommentForm';

describe('Comment Form', () => {
  it('should render correctly with props', () => {
    const output = shallow(
      <CommentForm name="Bob Loblaw" text="Rock on." />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
