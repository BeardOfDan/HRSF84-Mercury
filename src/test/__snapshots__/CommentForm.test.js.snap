// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Form should render correctly 1`] = `
<div
  className="form-comment container"
>
  <form>
    <input
      name="name"
      style={
        Object {
          "border": "1px solid black",
          "margin": "5px",
          "padding": "5px",
        }
      }
      type="text"
      value="Bob Loblaw"
    />
    <input
      name="comment"
      style={
        Object {
          "border": "1px solid black",
          "margin": "5px",
          "padding": "5px",
        }
      }
      type="text"
      value="Rock on."
    />
    <input
      type="submit"
      value="Submit Comment"
    />
  </form>
</div>
`;

exports[`Comment Form should render correctly with props 1`] = `
<div
  className="form-comment container"
>
  <form>
    <input
      name="name"
      style={
        Object {
          "border": "1px solid black",
          "margin": "5px",
          "padding": "5px",
        }
      }
      type="text"
      value="Bob Loblaw"
    />
    <input
      name="comment"
      style={
        Object {
          "border": "1px solid black",
          "margin": "5px",
          "padding": "5px",
        }
      }
      type="text"
      value="Rock on."
    />
    <input
      type="submit"
      value="Submit Comment"
    />
  </form>
</div>
`;
